src/services/hasura/hasura.service.ts

```
@@ -68,27 +68,48 @@ export class HasuraService {
-          return false;
+    // Utility to parse and evaluate a tag's value against a filter
+    const evaluateCondition = (tagValueJson, filterKey, filterValue) => {
+      try {
+        const tagValue = JSON.parse(tagValueJson);
+        const condition = tagValue.condition;
+        const conditionValues = tagValue.conditionValues;
+
+        if (filterKey === 'annualIncome' && filterValue.includes('-')) {
+          // Special case: Handle range for annualIncome
+          const [filterMin, filterMax] = filterValue.split('-').map(Number);
+          if (isNaN(filterMin) || isNaN(filterMax)) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)